payment/key.go

```diff
@@ -77,3 +83,65 @@ func GenerateSecret() (secret string, nonce string, err error) {
 
 	return fmt.Sprintf("%x", encryptedBytes), fmt.Sprintf("%x", nonceBytes), err
 }
+
+// LookupPublicKey returns the merchant key corresponding to the keyID used for verifying requests
+func (service *Service) LookupPublicKey(ctx context.Context, keyID string) (*httpsignature.Verifier, error) {
+	rootKeyIDStr, caveats, err := cryptography.DecodeKeyID(keyID)
+	if err != nil {
+		return nil, err
+	}
+
+	rootKeyID, err := uuid.FromString(rootKeyIDStr)
+	if err != nil {
+		return nil, fmt.Errorf("root key id must be a uuid: %v", err)
+	}
+
+	key, err := service.Datastore.GetKey(rootKeyID, false)
+	if err != nil {
+		return nil, err
+	}
+
+	secretKeyStr := key.SecretKey
```

Comment:
I think you will need to call 
```go
key.SetSecretKey()
secretKeyStr := key.SecretKey
```
as the key's "SecretKey" value will be an empty string until this is called.

Reply:
good catch. I don't love that there's no safeguard against this - what do you think about changing `SecretKey`'s type to `*string`?

Reply:
or better yet should we just make a getter and get rid of the on-struct version?

Reply:
I like your second idea better personally.